bundles/org.openhab.binding.thekeys/src/main/java/org/openhab/binding/thekeys/internal/gateway/TheKeysGatewayHandler.java

```diff
@@ -0,0 +1,152 @@
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.thekeys.internal.api.GatewayService;
+import org.openhab.binding.thekeys.internal.api.TheKeysHttpClient;
+import org.openhab.binding.thekeys.internal.api.model.GatewayInfosDTO;
+import org.openhab.binding.thekeys.internal.api.model.LockerDTO;
+import org.openhab.binding.thekeys.internal.provider.TheKeyTranslationProvider;
+import org.openhab.binding.thekeys.internal.smartlock.TheKeysDiscoveryService;
+import org.openhab.binding.thekeys.internal.smartlock.TheKeysSmartlockHandler;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TheKeysGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jordan Martin - Initial contribution
+ */
+@NonNullByDefault
+public class TheKeysGatewayHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TheKeysGatewayHandler.class);
+    private final TheKeyTranslationProvider translationProvider;
+    private @Nullable GatewayService api;
+
+    private @Nullable ScheduledFuture<?> gwPollingJob;
+
+    public TheKeysGatewayHandler(Bridge bridge, TheKeyTranslationProvider translationProvider) {
+        super(bridge);
+        this.translationProvider = translationProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // do nothing
+    }
+
+    @Override
+    public void initialize() {
+        TheKeysGatewayConfiguration config = getConfigAs(TheKeysGatewayConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NOT_YET_READY,
+                "@text/message.status.wait-data-from-gateway");
+        api = new GatewayService(config, new TheKeysHttpClient());
+        gwPollingJob = scheduler.scheduleWithFixedDelay(this::pollGateway, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Fetch the state of locks and notify the corresponding thing handler
+     */
+    private void pollGateway() {
+        try {
+            GatewayInfosDTO gwInfos = api.getGatewayInfos();
+            updateProperty(PROPERTY_VERSION, String.valueOf(gwInfos.getVersion()));
```

Comment:
Rather use `ThingType.PROPERTY_FIRMWARE_VERSION` in case this is a firmware version.